halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/DeployService.java

```diff
@@ -83,95 +88,92 @@ public NodeDiff configDiff(String deploymentName) {
+    SpinnakerServiceProvider serviceProvider = serviceProviderFactory.create(deploymentConfiguration);
+
+    if (serviceNames.isEmpty()) {
+      serviceNames = serviceProvider
+          .getServices()
+          .stream()
+          .map(SpinnakerService::getCannonicalName)
+          .collect(Collectors.toList());
+    }
+
+    ResolvedConfiguration resolvedConfiguration;
+    if (deployOptions.contains(DeployOption.OMIT_CONFIG)) {
+      resolvedConfiguration = generateService.generateConfig(deploymentName, Collections.emptyList());
+    } else {
+      resolvedConfiguration = generateService.generateConfig(deploymentName, serviceNames);
+    }
+
+    Deployer deployer = getDeployer(deploymentConfiguration);
+    DeploymentDetails deploymentDetails = getDeploymentDetails(deploymentConfiguration);
+
+    RemoteAction action = deployer.deploy(serviceProvider, deploymentDetails, resolvedConfiguration, serviceNames);
+    action.commitScript(halconfigDirectoryStructure.getInstallScriptPath(deploymentName));
+    return action;
+  }
+
+  private Deployer getDeployer(DeploymentConfiguration deploymentConfiguration) {
     DeploymentEnvironment.DeploymentType type = deploymentConfiguration.getDeploymentEnvironment().getType();
     switch (type) {
       case LocalDebian:
-        throw new HalException(FATAL, "Rollback for debians has not been implemented yet.");
+        return localDeployer;
       case Distributed:
-        rollbackDistributed(deploymentName);
+        return distributedDeployer;
       default:
         throw new IllegalArgumentException("Unrecognized deployment type " + type);
     }
   }
 
-  public RemoteAction deploy(String deploymentName) {
-    DeploymentConfiguration deploymentConfiguration = deploymentService.getDeploymentConfiguration(deploymentName);
+  private DeploymentDetails getDeploymentDetails(DeploymentConfiguration deploymentConfiguration) {
+    String deploymentName = deploymentConfiguration.getName();
+    BillOfMaterials billOfMaterials = artifactService.getBillOfMaterials(deploymentName);
     DeploymentEnvironment.DeploymentType type = deploymentConfiguration.getDeploymentEnvironment().getType();
     switch (type) {
       case LocalDebian:
-        return installSpinnaker(deploymentName);
+        return new DeploymentDetails()
+            .setDeploymentConfiguration(deploymentConfiguration)
+            .setDeploymentName(deploymentName)
+            .setBillOfMaterials(billOfMaterials);
       case Distributed:
-        return deploySpinnaker(deploymentName);
+        DeploymentEnvironment deploymentEnvironment = deploymentConfiguration.getDeploymentEnvironment();
+        String accountName = deploymentEnvironment.getAccountName();
+
+        if (accountName == null || accountName.isEmpty()) {
+          throw new HalException(FATAL, "An account name must be "
+              + "specified as the desired place to run your simple clustered deployment.");
```

Comment:
I think switching it to `actuate` sounds funny, and `run` conveys that this is where spinnaker is currently deployed, but maybe `deploy` is best.